packages/vkui/src/lib/children.test.tsx

```diff
@@ -0,0 +1,211 @@
+import { render, screen } from '@testing-library/react';
+import { childToString, getTextFromChildren, hasChildren } from './children';
+
+const OnlyText: React.FC<React.PropsWithChildren> = ({ children }) => (
+  <div data-testid="onlytext">{getTextFromChildren(children)}</div>
+);
+
+describe('children utils', () => {
+  describe('hasChildren', () => {
+    describe('returns true', () => {
+      it('with elements', () => {
+        const element = (
+          <>
+            <span />
+            <span />
+          </>
+        );
+
+        expect(hasChildren(element)).toBe(true);
+      });
+
+      it('with text', () => {
+        const element = <>My test</>;
+
+        expect(hasChildren(element)).toBe(true);
+      });
+
+      it('with numbers', () => {
+        const element = <div>{1}</div>;
+
+        expect(hasChildren(element)).toBe(true);
+      });
+
+      it('with true', () => {
+        const element = <div>{true}</div>;
+
+        expect(hasChildren(element)).toBe(true);
+      });
+
+      it('with combined types', () => {
+        const element = (
+          <>
+            <span />
+            My test
+            {1}
+          </>
+        );
+
+        expect(hasChildren(element)).toBe(true);
+      });
+
+      it('with valid and non valid types', () => {
+        const element = (
+          <>
+            <span />
+            My test
+            {null}
+            {false}
+          </>
+        );
+
+        expect(hasChildren(element)).toBe(true);
+      });
+    });
+
+    describe('returns false', () => {
+      it('when empty', () => {
+        const element = <div />;
+
+        expect(hasChildren(element)).toBe(false);
+      });
+
+      it('with null', () => {
+        const element = <div>{null}</div>;
+
+        expect(hasChildren(element)).toBe(false);
+      });
+
+      it('with false', () => {
+        const element = <div>{null}</div>;
+
+        expect(hasChildren(element)).toBe(false);
+      });
+    });
+  });
+
+  describe('childToString', () => {
+    it('string', () => {
+      expect(childToString('a')).toBe('a');
+    });
+
+    it('number', () => {
+      expect(childToString(1)).toBe('1');
+    });
+
+    it('boolean', () => {
+      expect(childToString(true)).toBe('');
+    });
+
+    it('{}', () => {
+      // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+      expect(childToString({} as React.ReactNode)).toBe('');
+    });
+
+    it('null', () => {
+      expect(childToString(null)).toBe('');
+    });
+
+    it('undefined', () => {
+      expect(childToString()).toBe('');
+    });
+  });
+
+  describe('getTextFromChildren', () => {
+    it('on nested elements', () => {
+      render(
+        <OnlyText>
+          <span>0</span>
+          <b>1</b>
+          <span>
+            <i>2</i>
+          </span>
+          <i>3</i>
+        </OnlyText>,
+      );
+
+      expect(screen.getByTestId('onlytext').textContent).toBe('0123');
```

Comment:
```suggestion
      expect(screen.getByTestId('onlytext')).toHaveTextContent('0123');
```
вот так можно

вроде x)

ну и ниже в 4 строчках